documentation/hpcc-tf-for-developers.md

```diff
@@ -0,0 +1,373 @@
+
+## Root Directory
+Here is the root directory's contents (<font color="blue">**blue**</font> names are subdirectories) and a description of each entry:
+
+
+|Entry Name|Description|
+|:-----|:----------|
+| `lite-variables.tf` | Contains all input variables |
+| `lite.auto.tfvars.example` |Is an example .auto.tfvars file |
+| `main.tf` | Contains most of the terraform that deploys all components of system |
+| `providers.tf` | Contains one provider, azurerm |
+| <font color="blue">`scripts`</font> | Directory containing scripts used in deployment |
+| <font color="blue">`aks`</font> | Directory containing terraform to deploy `aks` |
+| <font color="blue">`hpcc`</font> | Directory containing terraform to deploy `hpcc` |
+| <font color="blue">`storage`</font> | Directory containing terraform to deploy external or persistent `storage` |
+| <font color="blue">`vnet`</font> | Directory containing terraform to deploy virtual network used by `aks` |
+
+
+The following table shows all the variables in the file, `lite-variables.tf`, and their types. Plus, the table gives a description of each variable. Also, when one deploys from the root directory the `deploy` script puts these variables (or some of them) in the subdirectory where the deployment takes place.
+
+|Variable|Type|Description|
+|:-----|:---|:----------|
+| `admin_username` | string | Username of the administrator of this HPCC Systems cluster. Example entry: "jdoe" |
+| `aks_admin_email` | string | Email address of the administrator of this HPCC Systems cluster. Example entry: "jane.doe@hpccsystems.com" |
+| `aks_admin_ip_cidr_map` | map of string | Map of name => CIDR IP addresses that can administrate this AKS. Format is '{"name"="cidr" [, "name"="cidr"]*}'. The 'name' portion must be unique. To add no CIDR addresses, use '{}'. The corporate network and your current IP address will be added automatically, and these addresses will have access to the HPCC cluster as a user. |
+| `aks_admin_name` | string | Name of the administrator of this HPCC Systems cluster. Example entry: "Jane Doe" |
+| `aks_azure_region` | string | The Azure region abbreviation in which to create these resources. Must be one of ["eastus", "eastus2", "centralus"]. Example entry: "eastus" |
+| `aks_dns_zone_name` | string | Name of an existing dns zone. Example entry: "hpcczone.us-hpccsystems-dev.azure.lnrsg.io" |
+| `aks_dns_zone_resource_group_name` | string | Name of the resource group of the above dns zone. Example entry: "app-dns-prod-eastus2" |
+| `aks_enable_roxie` | boolean | Enable ROXIE? This will also expose port 8002 on the cluster. Example entry: false |
+| `aks_max_node_count` | number | The maximum number of VM nodes to allocate for the HPCC Systems node pool. Must be 2 or more. |
+| `aks_node_size` | string | The VM size for each node in the HPCC Systems node pool. Recommend "Standard_B4ms" or better. See https://docs.microsoft.com/en-us/azure/virtual-machines/sizes-general for more information. |
```

Comment:
Is the description here still accurate?  I thought you were no longer using Azure VM sizes (using 'large' and 'xlarge' and such instead).